common/spec/spec_helper.rb

```diff
@@ -1,14 +1,32 @@
-# typed: false
+# typed: true
 # frozen_string_literal: true
 
 require "rspec/its"
 require "rspec/sorbet"
 require "webmock/rspec"
 require "vcr"
 require "debug"
+require "simplecov"
+require "simplecov_json_formatter"
 require "stackprof"
 require "uri"
 
+# SimpleCov _must_ be started before any dependabot code is loaded
+SimpleCov.start do
+  add_filter "/spec/"
+  if ENV["CI"]
+    formatter SimpleCov::Formatter::SimpleFormatter
+  else
+    formatter SimpleCov::Formatter::MultiFormatter.new([
+      SimpleCov::Formatter::SimpleFormatter,
+      SimpleCov::Formatter::HTMLFormatter
```

Comment:
I understand the sentiment here, but I think that we should still generate the HTML report during the CI build and consider adding them as a build artifact for easy access

Reply:
Fair point. I'll change that and use `actions/updload-artifact`.

Reply:
@yeikel looks like this was merged. I'll leave #9203 open and handle it in a follow-up PR.